packages/server/src/api/controllers/public/applications.ts

```diff
@@ -0,0 +1,71 @@
+const { getAllApps } = require("@budibase/backend-core/db")
+const { updateAppId } = require("@budibase/backend-core/context")
+import { search as stringSearch } from "./utils"
+import { default as controller } from "../application"
+import { Application } from "../../../definitions/common"
+
+function fixAppID(app: Application, params: any) {
+  if (!params) {
+    return app
+  }
+  if (!app._id && params.appId) {
+    app._id = params.appId
+  }
+  return app
+}
+
+async function setResponseApp(ctx: any) {
+  if (ctx.body && ctx.body.appId && (!ctx.params || !ctx.params.appId)) {
+    ctx.params = { appId: ctx.body.appId }
+  }
+  await controller.fetchAppPackage(ctx)
```

Comment:
Is there being returned in the response here than the docs suggest?
in `fetchAppPackage`
```js
  ctx.body = {
    application,
    screens,
    layouts,
    clientLibPath: clientLibraryPath(ctx.params.appId, application.version),
  }
```

Reply:
Yes, there is, but I didn't want to document the screens, layouts and client lib stuff as its not useful to the user right now. I thought though it would be good to return everything that we when we get an app through our internal API, as it means its consistent at least.

In the future we can document it, but I think we'd need to expose the ability to alter the screens/layouts through the public API as well before we expose/explain that extra stuff.